Core/Rewriters/Strategies/ConstraintsStrategies/ConstraintsRewriteStrategyFactory.cs

```diff
@@ -37,21 +39,87 @@ public static IConstraintsRewriteStrategy GetRewriteStrategy (ExpressionSyntax n
-          var s when rhinoMocksConstraintsIsSymbol.GetMembers ("LessThan").Contains (s) => IsLessThanConstraintsRewriteStrategy.Instance,
-          var s when rhinoMocksConstraintsIsSymbol.GetMembers ("LessThanOrEqual").Contains (s) => IsLessThanOrEqualConstraintsRewriteStrategy.Instance,
-          var s when rhinoMocksConstraintsListSymbol.GetMembers ("IsIn").Contains (s) => ListIsInConstraintsRewriteStrategy.Instance,
-          var s when rhinoMocksConstraintsListSymbol.GetMembers ("ContainsAll").Contains (s) => ListContainsAllConstraintsRewriteStrategy.Instance,
-          var s when rhinoMocksConstraintsPropertySymbol.GetMembers ("Value").Contains (s) => PropertyValueConstraintsRewriteStrategy.Instance,
+          _ when rhinoMocksConstraintsIsSymbol.GetMembers ("Equal").Contains (symbol)
+              => IsEqualConstraintsRewriteStrategy.Instance,
+          _ when rhinoMocksConstraintsIsSymbol.GetMembers ("NotEqual").Contains (symbol)
+              => IsNotEqualConstraintsRewriteStrategy.Instance,
+          _ when rhinoMocksConstraintsIsSymbol.GetMembers ("Same").Contains (symbol)
+              => IsSameConstraintsRewriteStrategy.Instance,
+          _ when rhinoMocksConstraintsIsSymbol.GetMembers ("NotSame").Contains (symbol)
+              => IsNotSameConstraintsRewriteStrategy.Instance,
+          _ when rhinoMocksConstraintsIsSymbol.GetMembers ("Null").Contains (symbol)
+              => IsNullConstraintsRewriteStrategy.Instance,
+          _ when rhinoMocksConstraintsIsSymbol.GetMembers ("NotNull").Contains (symbol)
+              => DetermineNotNullStrategy (node, model, rhinoMocksConstraintsPropertySymbol),
+          _ when rhinoMocksConstraintsIsSymbol.GetMembers ("GreaterThan").Contains (symbol)
+              => IsGreaterThanConstraintsRewriteStrategy.Instance,
+          _ when rhinoMocksConstraintsIsSymbol.GetMembers ("GreaterThanOrEqual").Contains (symbol)
+              => IsGreaterThanOrEqualConstraintsRewriteStrategy.Instance,
+          _ when rhinoMocksConstraintsIsSymbol.GetMembers ("LessThan").Contains (symbol)
+              => IsLessThanConstraintsRewriteStrategy.Instance,
+          _ when rhinoMocksConstraintsIsSymbol.GetMembers ("LessThanOrEqual").Contains (symbol)
+              => IsLessThanOrEqualConstraintsRewriteStrategy.Instance,
+          _ when rhinoMocksConstraintsListSymbol.GetMembers ("IsIn").Contains (symbol)
+              => ListIsInConstraintsRewriteStrategy.Instance,
+          _ when rhinoMocksConstraintsListSymbol.GetMembers ("ContainsAll").Contains (symbol)
+              => ListContainsAllConstraintsRewriteStrategy.Instance,
+          _ when rhinoMocksConstraintsPropertySymbol.GetMembers ("Value").Contains (symbol)
+              => DeterminePropertyValueStrategy (node, model, rhinoMocksConstraintsIsSymbol, rhinoMocksConstraintsPropertySymbol),
           _ => DefaultConstraintsRewriteStrategy.Instance
       };
```

Comment:
I think I understand what you're doing here (matching the only supported pattern of linked constraits). Wouldn't it be easier to have one method called for `Is.NotNull` & `Property.Value`, checking the structure and returning fitting expressions?

Reply:
you are right this would be much nicer and easier to understand